Non-uniform refinement fails at import half maps from relion

Hi!

There seems to be a problem with running non-uniform refinement when using a 3D reference from relion refinement (I'm using the relion 4.0 beta). The path to the half maps seems to be malformed.


This is the stdout:
```
00001:   Logging configured. STDOUT --> Runs/003182_ProtCryoSparcNewNonUniformRefine3D/logs/run.stdout , STDERR --> Runs/003182_ProtCryoSparcNewNonUniformRefine3D/logs/run.stderr
00002:   RUNNING PROTOCOL -----------------
00003:   Protocol starts
00004:   Hostname: em3
00005:   PID: 105038
00006:   pyworkflow: 3.0.20
00007:   plugin: cryosparc2
00008:   plugin v: 3.3.13
00009:   currentDir: /media/odin/genval/NPM_cryolo_WARP_2_low_salt
00010:   workingDir: Runs/003182_ProtCryoSparcNewNonUniformRefine3D
00011:   runMode: Restart
00012:             MPI: 1
00013:         threads: 1
00014:    Starting at step: 1
00015:    Running steps 
00016:   STARTED: convertInputStep, step 1, time 2022-05-05 11:11:18.121081
00017:   convertBinaryFiles: creating soft links.
00018:      Root: Runs/003182_ProtCryoSparcNewNonUniformRefine3D/tmp/input -> Runs/001051_CopyToScratch/extra/Runs
00019:   Importing particles...
00020:   Running: /home/genis/software/cryosparc/cryosparc2_master/bin/cryosparcm cli 'make_job("import_particles","P40","W26", "scipion", "None", "None", {"particle_meta_path": "/media/odin/genval/NPM_cryolo_WARP_2_low_salt/Runs/003182_ProtCryoSparcNewNonUniformRefine3D/tmp/input_particles.star", "particle_blob_path": "/media/odin/genval/NPM_cryolo_WARP_2_low_salt/Runs/003182_ProtCryoSparcNewNonUniformRefine3D/tmp", "psize_A": "1.01"}, {}, "False", 0)'
00021:   Got J67 for JobId
00022:   Running: /home/genis/software/cryosparc/cryosparc2_master/bin/cryosparcm cli 'enqueue_job("P40","J67","default", "em3", False, "False")'
00023:   Importing volume...
00024:   Running: /home/genis/software/cryosparc/cryosparc2_master/bin/cryosparcm cli 'make_job("import_volumes","P40","W26", "scipion", "None", "None", {"volume_blob_path": "/media/odin/genval/NPM_cryolo_WARP_2_low_salt/Runs/002101_ProtRelionRefine3D/extra/relion_class001.mrc", "volume_out_name": "map", "volume_psize": "1.01"}, {}, "False", 0)'
00025:   Got J68 for JobId
00026:   Running: /home/genis/software/cryosparc/cryosparc2_master/bin/cryosparcm cli 'enqueue_job("P40","J68","default", "em3", False, "False")'
00027:   Importing half volume A...
00028:   Running: /home/genis/software/cryosparc/cryosparc2_master/bin/cryosparcm cli 'make_job("import_volumes","P40","W26", "scipion", "None", "None", {"volume_blob_path": "/media/odin/genval/NPM_cryolo_WARP_2_low_salt/Runs/002101_ProtRelionRefine3D/extra/relion_half1_class001_unfil.mrc:mrc", "volume_out_name": "map_half_A", "volume_psize": "1.01"}, {}, "False", 0)'
00029:   Got J69 for JobId
00030:   Running: /home/genis/software/cryosparc/cryosparc2_master/bin/cryosparcm cli 'enqueue_job("P40","J69","default", "em3", False, "False")'
00031:   Protocol failed: An error occurred importing the volume. Please, go to cryoSPARC software for more details.
00032:   FAILED: convertInputStep, step 1, time 2022-05-05 11:17:36.259135
```

This is the stderror:
```
00001:   Traceback (most recent call last):
00002:     File "/home/genis/anaconda3/envs/scipion3/lib/python3.8/site-packages/pyworkflow/protocol/protocol.py", line 201, in run
00003:       self._run()
00004:     File "/home/genis/anaconda3/envs/scipion3/lib/python3.8/site-packages/pyworkflow/protocol/protocol.py", line 252, in _run
00005:       resultFiles = self._runFunc()
00006:     File "/home/genis/anaconda3/envs/scipion3/lib/python3.8/site-packages/pyworkflow/protocol/protocol.py", line 248, in _runFunc
00007:       return self._func(*self._args)
00008:     File "/home/genis/anaconda3/envs/scipion3/lib/python3.8/site-packages/cryosparc2/protocols/protocol_base.py", line 103, in convertInputStep
00009:       self._importVolume()
00010:     File "/home/genis/anaconda3/envs/scipion3/lib/python3.8/site-packages/cryosparc2/protocols/protocol_base.py", line 214, in _importVolume
00011:       importVolumeHalfAJob = doImportVolumes(self, map_half_A_fn, vol,
00012:     File "/home/genis/anaconda3/envs/scipion3/lib/python3.8/site-packages/cryosparc2/utils.py", line 332, in doImportVolumes
00013:       waitForCryosparc(protocol.projectName.get(), importedVolume.get(),
00014:     File "/home/genis/anaconda3/envs/scipion3/lib/python3.8/site-packages/cryosparc2/utils.py", line 471, in waitForCryosparc
00015:       raise Exception(failureMessage)
00016:   Exception: An error occurred importing the volume. Please, go to cryoSPARC software for more details.
00017:   Protocol failed: An error occurred importing the volume. Please, go to cryoSPARC software for more details.h
```

Cryosparc error:
```
[CPU: 243.4 MB]  Traceback (most recent call last):
  File "cryosparc_master/cryosparc_compute/run.py", line 85, in cryosparc_compute.run.main
  File "/home/genis/software/cryosparc/cryosparc2_master/cryosparc_compute/jobs/imports/run.py", line 580, in run_import_volumes
    assert len(allpaths) > 0, "No paths match!"
AssertionError: No paths match!
```


My plugin version, when I run
`scipion3 installp -p scipion-em-cryosparc2`
I get
```
Building scipion-em-cryosparc2 ...
/home/genis/anaconda3/envs/scipion3/bin/python -m pip install scipion-em-cryosparc2==3.3.13
Requirement already satisfied: scipion-em-cryosparc2==3.3.13 in ./anaconda3/envs/scipion3/lib/python3.8/site-packages (3.3.13)
Requirement already satisfied: numba<=0.47.0 in ./anaconda3/envs/scipion3/lib/python3.8/site-packages (from scipion-em-cryosparc2==3.3.13) (0.47.0)
Requirement already satisfied: pandas<=1.3.5 in ./anaconda3/envs/scipion3/lib/python3.8/site-packages (from scipion-em-cryosparc2==3.3.13) (1.2.2)
Requirement already satisfied: scipy>=0.17.0 in ./anaconda3/envs/scipion3/lib/python3.8/site-packages (from scipion-em-cryosparc2==3.3.13) (1.6.1)
Requirement already satisfied: llvmlite<0.32.0 in ./anaconda3/envs/scipion3/lib/python3.8/site-packages (from scipion-em-cryosparc2==3.3.13) (0.31.0)
Requirement already satisfied: emtable in ./anaconda3/envs/scipion3/lib/python3.8/site-packages (from scipion-em-cryosparc2==3.3.13) (0.0.10)
Requirement already satisfied: numpy in ./anaconda3/envs/scipion3/lib/python3.8/site-packages (from numba<=0.47.0->scipion-em-cryosparc2==3.3.13) (1.18.4)
Requirement already satisfied: setuptools in ./anaconda3/envs/scipion3/lib/python3.8/site-packages (from numba<=0.47.0->scipion-em-cryosparc2==3.3.13) (60.6.0)
Requirement already satisfied: pytz>=2017.3 in ./anaconda3/envs/scipion3/lib/python3.8/site-packages (from pandas<=1.3.5->scipion-em-cryosparc2==3.3.13) (2021.3)
Requirement already satisfied: python-dateutil>=2.7.3 in ./anaconda3/envs/scipion3/lib/python3.8/site-packages (from pandas<=1.3.5->scipion-em-cryosparc2==3.3.13) (2.8.2)
Requirement already satisfied: six>=1.5 in ./anaconda3/envs/scipion3/lib/python3.8/site-packages (from python-dateutil>=2.7.3->pandas<=1.3.5->scipion-em-cryosparc2==3.3.13) (1.16.0)
Done (1.00 seconds)
Building pyem-22.01.18 ...
  All targets exist, skipping.
Done (0.00 seconds)
```

--- a/relion/convert/__init__.py
+++ b/relion/convert/__init__.py
@@ -148,7 +148,7 @@ class ClassesLoader:
             index, fn, row = self._classesInfo[classId]
             item.setAlignment(self._alignType)
             if self._alignType == ALIGN_PROJ:
-                fn += ':mrc'  # mark reference as a MRC volume
+                fn += ':mrc'  # FIXME: mark reference as a MRC volume
             item.getRepresentative().setLocation(index, fn)
             item._rlnClassDistribution = Float(row.rlnClassDistribution)
             item._rlnAccuracyRotations = Float(row.rlnAccuracyRotations)

--- a/relion/convert/dataimport.py
+++ b/relion/convert/dataimport.py
@@ -130,7 +130,7 @@ class RelionImport:
         classId = item.getObjId()
         if classId in self._classesDict:
             index, fn, row = self._classesDict[classId]
-            if fn.endswith('.mrc'):
+            if fn.endswith('.mrc'):  # FIXME
                 fn += ':mrc'  # Specify that are volumes to read them properly in xmipp
             item.getRepresentative().setLocation(index, fn)
             item._rlnclassDistribution = Float(row.get('rlnClassDistribution'))

--- a/relion/protocols/protocol_base.py
+++ b/relion/protocols/protocol_base.py
@@ -52,7 +52,7 @@ class ProtRelionBase(EMProtocol):
     line and the files produced.
 
     Most of the Relion protocols, have two modes: NORMAL or CONTINUE. That's why
-    some of the function have a template pattern approach to define the behaviour
+    some functions have a template pattern approach to define the behaviour
     depending on the case.
     """
     _label = None
@@ -73,13 +73,13 @@ class ProtRelionBase(EMProtocol):
         EMProtocol.__init__(self, **args)
 
     def _initialize(self):
-        """ This function is mean to be called after the
+        """ This function is meant to be called after the
         working dir for the protocol have been set. (maybe after recovery from mapper)
         """
         self._createFilenameTemplates()
         self._createIterTemplates()
 
-        self.ClassFnTemplate = '%(rootDir)s/relion_it%(iter)03d_class%(ref)03d.mrc:mrc'
+        self.ClassFnTemplate = '%(rootDir)s/relion_it%(iter)03d_class%(ref)03d.mrc'
         if not self.doContinue:
             self.continueRun.set(None)
         else:
@@ -104,16 +104,16 @@ class ProtRelionBase(EMProtocol):
             'selected_volumes': self._getPath('selected_volumes_xmipp.xmd'),
             'dataFinal': self._getExtraPath("relion_data.star"),
             'modelFinal': self._getExtraPath("relion_model.star"),
-            'finalvolume': self._getExtraPath("relion_class%(ref3d)03d.mrc:mrc"),
-            'final_half1_volume': self._getExtraPath("relion_half1_class%(ref3d)03d_unfil.mrc:mrc"),
-            'final_half2_volume': self._getExtraPath("relion_half2_class%(ref3d)03d_unfil.mrc:mrc"),
-            'finalSGDvolume': self._getExtraPath("relion_it%(iter)03d_class%(ref3d)03d.mrc:mrc"),
+            'finalvolume': self._getExtraPath("relion_class%(ref3d)03d.mrc"),
+            'final_half1_volume': self._getExtraPath("relion_half1_class%(ref3d)03d_unfil.mrc"),
+            'final_half2_volume': self._getExtraPath("relion_half2_class%(ref3d)03d_unfil.mrc"),
+            'finalSGDvolume': self._getExtraPath("relion_it%(iter)03d_class%(ref3d)03d.mrc"),
             'preprocess_particles': self._getPath("preprocess_particles.mrcs"),
             'preprocess_particles_star': self._getPath("preprocess_particles.star"),
             'preprocess_particles_prefix': "preprocess_particles",
-            'finalvolume_mbody': self._getExtraPath("relion_body%(ref3d)03d.mrc:mrc"),
-            'final_half1_volume_mbody': self._getExtraPath("relion_half1_body%(ref3d)03d_unfil.mrc:mrc"),
-            'final_half2_volume_mbody': self._getExtraPath("relion_half2_body%(ref3d)03d_unfil.mrc:mrc"),
+            'finalvolume_mbody': self._getExtraPath("relion_body%(ref3d)03d.mrc"),
+            'final_half1_volume_mbody': self._getExtraPath("relion_half1_body%(ref3d)03d_unfil.mrc"),
+            'final_half2_volume_mbody': self._getExtraPath("relion_half2_body%(ref3d)03d_unfil.mrc"),
         }
         # add to keys, data.star, optimiser.star and sampling.star
         for key in self.FILE_KEYS:
@@ -123,8 +123,8 @@ class ProtRelionBase(EMProtocol):
         # add other keys that depends on prefixes
         for p in self.PREFIXES:
             myDict['%smodel' % p] = self.extraIter + '%smodel.star' % p
-            myDict['%svolume' % p] = self.extraIter + p + 'class%(ref3d)03d.mrc:mrc'
-            myDict['%svolume_mbody' % p] = self.extraIter + p + 'body%(ref3d)03d.mrc:mrc'
+            myDict['%svolume' % p] = self.extraIter + p + 'class%(ref3d)03d.mrc'
+            myDict['%svolume_mbody' % p] = self.extraIter + p + 'body%(ref3d)03d.mrc'
 
         self._updateFilenamesDict(myDict)
 

--- a/relion/protocols/protocol_ctf_refinement.py
+++ b/relion/protocols/protocol_ctf_refinement.py
@@ -51,20 +51,20 @@ class ProtRelionCtfRefinement(ProtParticles, ProtRelionBase):
         myDict = {
             'output_star': self._getExtraPath("particles_ctf_refine.star"),
             'ctf_sqlite': self._getExtraPath("ctf_analyze.sqlite"),
-            'mag_obs_x': self._getExtraPath("mag_disp_x_optics-group_%(og)d.mrc:mrc"),
-            'mag_obs_y': self._getExtraPath("mag_disp_y_optics-group_%(og)d.mrc:mrc"),
-            'mag_fit_x': self._getExtraPath("mag_disp_x_fit_optics-group_%(og)d.mrc:mrc"),
-            'mag_fit_y': self._getExtraPath("mag_disp_y_fit_optics-group_%(og)d.mrc:mrc"),
-            'tetrafoil_it_fit': self._getExtraPath("aberr_delta-phase_iter-fit_optics-group_%(og)d_N-4.mrc:mrc"),
-            'tetrafoil_fit': self._getExtraPath("aberr_delta-phase_lin-fit_optics-group_%(og)d_N-4.mrc:mrc"),
-            'tetrafoil_residual_fit': self._getExtraPath("aberr_delta-phase_lin-fit_optics-group_%(og)d_N-4_residual.mrc:mrc"),
-            'tetrafoil_obs': self._getExtraPath("aberr_delta-phase_per-pixel_optics-group_%(og)d.mrc:mrc"),
-            'beamtilt_it_fit': self._getExtraPath("beamtilt_delta-phase_iter-fit_optics-group_%(og)d.mrc:mrc"),
-            'beamtilt_fit': self._getExtraPath("beamtilt_delta-phase_lin-fit_optics-group_%(og)d.mrc:mrc"),
-            'trefoil_it_fit': self._getExtraPath("beamtilt_delta-phase_iter-fit_optics-group_%(og)d_N-3.mrc:mrc"),
-            'trefoil_fit': self._getExtraPath("beamtilt_delta-phase_lin-fit_optics-group_%(og)d_N-3.mrc:mrc"),
-            'trefoil_residual_fit': self._getExtraPath("beamtilt_delta-phase_lin-fit_optics-group_%(og)d_N-3_residual.mrc:mrc"),
-            'beamtilt_obs': self._getExtraPath("beamtilt_delta-phase_per-pixel_optics-group_%(og)d.mrc:mrc")
+            'mag_obs_x': self._getExtraPath("mag_disp_x_optics-group_%(og)d.mrc"),
+            'mag_obs_y': self._getExtraPath("mag_disp_y_optics-group_%(og)d.mrc"),
+            'mag_fit_x': self._getExtraPath("mag_disp_x_fit_optics-group_%(og)d.mrc"),
+            'mag_fit_y': self._getExtraPath("mag_disp_y_fit_optics-group_%(og)d.mrc"),
+            'tetrafoil_it_fit': self._getExtraPath("aberr_delta-phase_iter-fit_optics-group_%(og)d_N-4.mrc"),
+            'tetrafoil_fit': self._getExtraPath("aberr_delta-phase_lin-fit_optics-group_%(og)d_N-4.mrc"),
+            'tetrafoil_residual_fit': self._getExtraPath("aberr_delta-phase_lin-fit_optics-group_%(og)d_N-4_residual.mrc"),
+            'tetrafoil_obs': self._getExtraPath("aberr_delta-phase_per-pixel_optics-group_%(og)d.mrc"),
+            'beamtilt_it_fit': self._getExtraPath("beamtilt_delta-phase_iter-fit_optics-group_%(og)d.mrc"),
+            'beamtilt_fit': self._getExtraPath("beamtilt_delta-phase_lin-fit_optics-group_%(og)d.mrc"),
+            'trefoil_it_fit': self._getExtraPath("beamtilt_delta-phase_iter-fit_optics-group_%(og)d_N-3.mrc"),
+            'trefoil_fit': self._getExtraPath("beamtilt_delta-phase_lin-fit_optics-group_%(og)d_N-3.mrc"),
+            'trefoil_residual_fit': self._getExtraPath("beamtilt_delta-phase_lin-fit_optics-group_%(og)d_N-3_residual.mrc"),
+            'beamtilt_obs': self._getExtraPath("beamtilt_delta-phase_per-pixel_optics-group_%(og)d.mrc")
         }
 
         self._updateFilenamesDict(myDict)

--- a/relion/viewers/viewer_base.py
+++ b/relion/viewers/viewer_base.py
@@ -510,12 +510,12 @@ Examples:
         files = []
         volumes = self._getVolumeNames()
         for volFn in volumes:
-            if not os.path.exists(volFn.replace(':mrc', '')):
+            if not os.path.exists(volFn):
                 raise Exception("Missing volume file: %s\n Please select "
                                 "a valid class or iteration number."
                                 % volFn)
             print("Adding vol: %s" % volFn)
-            files.append(volFn)
+            files.append(volFn + ":mrc")
 
         self.createVolumesSqlite(files, path, samplingRate)
         return [ObjectView(self._project, self.protocol.strId(), path)]
@@ -526,8 +526,6 @@ Examples:
         cmdFile = self.protocol._getExtraPath('chimera_volumes.cxc')
         with open(cmdFile, 'w+') as f:
             for vol in volumes:
-                # remove ":mrc" extension needed by xmipp
-                vol = vol.replace(":mrc", "")
                 # We assume that the chimera script will be generated
                 # at the same folder as relion volumes
                 localVol = os.path.basename(vol)
@@ -567,7 +565,7 @@ Examples:
         # If just one reference we can show the angular distribution
         ref3d = self._refsList[0]
         volFn = self._getVolumeNames()[0]
-        if not os.path.exists(volFn.replace(":mrc", "")):
+        if not os.path.exists(volFn):
             raise Exception("This class is empty. Please try with another class")
 
         for prefix in prefixes:
@@ -859,7 +857,7 @@ Examples:
                 for prefix in prefixes:
                     volFn = self.protocol._getFileName(prefix + 'volume',
                                                        iter=it, ref3d=ref3d)
-                    if os.path.exists(volFn.replace(':mrc', '')):
+                    if os.path.exists(volFn):
                         vols.append(volFn)
                     else:
                         raise Exception("Volume %s does not exists. \n"

--- a/relion/viewers/viewer_base.py
+++ b/relion/viewers/viewer_base.py
@@ -515,7 +515,8 @@ Examples:
                                 "a valid class or iteration number."
                                 % volFn)
             print("Adding vol: %s" % volFn)
-            files.append(volFn + ":mrc")
+            if not volFn.endswith(":mrc"):
+                files.append(volFn + ":mrc")
 
         self.createVolumesSqlite(files, path, samplingRate)
         return [ObjectView(self._project, self.protocol.strId(), path)]
@@ -529,6 +530,7 @@ Examples:
                 # We assume that the chimera script will be generated
                 # at the same folder as relion volumes
                 localVol = os.path.basename(vol)
+                vol = vol.replace(":mrc", "")
                 if os.path.exists(vol):
                     f.write("open %s\n" % localVol)
             f.write('tile\n')
@@ -858,6 +860,7 @@ Examples:
                     volFn = self.protocol._getFileName(prefix + 'volume',
                                                        iter=it, ref3d=ref3d)
                     if os.path.exists(volFn):
+                        volFn = volFn.replace(":mrc", "")
                         vols.append(volFn)
                     else:
                         raise Exception("Volume %s does not exists. \n"

--- a/relion/viewers/viewer_ctfrefine.py
+++ b/relion/viewers/viewer_ctfrefine.py
@@ -195,9 +195,6 @@ class ProtCtfRefineViewer(ProtocolViewer):
             img = ih.read(imgFn)
             ax.imshow(img.getData(), cmap='jet')
 
-            # with mrcfile.open(img.replace(":mrc", "")) as mrc:
-            #    im = ax.imshow(mrc.data, cmap='jet')
-
         return [plotter]
 
     def _displayBeamTilt(self, param=None):

--- a/relion/viewers/viewer_multibody.py
+++ b/relion/viewers/viewer_multibody.py
@@ -97,7 +97,8 @@ class MultibodyViewer(RelionViewer):
                                 "a valid class or iteration number."
                                 % volFn)
             print("Adding vol: %s" % volFn)
-            files.append(volFn + ":mrc")
+            if not volFn.endswith(":mrc"):
+                files.append(volFn + ":mrc")
 
         self.createVolumesSqlite(files, path, samplingRate)
         return [ObjectView(self._project, self.protocol.strId(), path)]
@@ -113,6 +114,7 @@ class MultibodyViewer(RelionViewer):
                     volFn = self.protocol._getFileName(prefix + 'volume_mbody',
                                                        iter=it, ref3d=ref3d)
                     if os.path.exists(volFn):
+                        volFn = volFn.replace(":mrc", "")
                         vols.append(volFn)
                     else:
                         raise Exception("Volume %s does not exists. \n"

--- a/relion/viewers/viewer_multibody.py
+++ b/relion/viewers/viewer_multibody.py
@@ -85,19 +85,19 @@ class MultibodyViewer(RelionViewer):
                 }
 
     def _createVolumesSqlite(self):
-        """ Write an sqlite with all volumes selected for visualization. """
+        """ Write a sqlite with all volumes selected for visualization. """
         path = self.protocol._getExtraPath('relion_viewer_volumes.sqlite')
         samplingRate = self.protocol.protRefine.get()._getInputParticles().getSamplingRate()
 
         files = []
         volumes = self._getVolumeNames()
         for volFn in volumes:
-            if not os.path.exists(volFn.replace(':mrc', '')):
+            if not os.path.exists(volFn):
                 raise Exception("Missing volume file: %s\n Please select "
                                 "a valid class or iteration number."
                                 % volFn)
             print("Adding vol: %s" % volFn)
-            files.append(volFn)
+            files.append(volFn + ":mrc")
 
         self.createVolumesSqlite(files, path, samplingRate)
         return [ObjectView(self._project, self.protocol.strId(), path)]
@@ -112,7 +112,7 @@ class MultibodyViewer(RelionViewer):
                 for prefix in prefixes:
                     volFn = self.protocol._getFileName(prefix + 'volume_mbody',
                                                        iter=it, ref3d=ref3d)
-                    if os.path.exists(volFn.replace(':mrc', '')):
+                    if os.path.exists(volFn):
                         vols.append(volFn)
                     else:
                         raise Exception("Volume %s does not exists. \n"

--- a/relion/viewers/viewer_postprocess.py
+++ b/relion/viewers/viewer_postprocess.py
@@ -114,7 +114,7 @@ class PostprocessViewer(ProtocolViewer):
         return [view]
 
     def _showVolume(self, paramName=None):
-        volPath = self.protocol._getExtraPath('postprocess.mrc:mrc')
+        volPath = self.protocol._getExtraPath('postprocess.mrc')
 
         if self.displayVol == VOLUME_CHIMERA:
             return self._showVolumesChimera(volPath)
@@ -123,7 +123,7 @@ class PostprocessViewer(ProtocolViewer):
             return self._showVolumeShowj(volPath)
 
     def _showMaskedVolume(self, paramName=None):
-        volPath = self.protocol._getExtraPath('postprocess_masked.mrc:mrc')
+        volPath = self.protocol._getExtraPath('postprocess_masked.mrc')
 
         if self.displayMaskedVol == VOLUME_CHIMERA:
             return self._showVolumesChimera(volPath)

